### ISSUE TITLE
Error executing one or more requests.

### ISSUE THREAD

**Issue by tesmez** [BurningOKR/BurningOKR#261]
Hello Support Team,
I have face with the below error while i try to creat OKRTeam and KeyResult on BurningOKR Portal 
Please help me how to resolve this issue

javax.persistence.RollbackException: Error while committing the transaction
org.hibernate.AssertionFailure: Unable to perform beforeTransactionCompletion callback
java.lang.NullPointerException
javax.persistence.RollbackException: Error while committing the transaction
org.hibernate.AssertionFailure: Unable to perform beforeTransactionCompletion callback
java.lang.NullPointerException
![image](https://github.com/BurningOKR/BurningOKR/assets/66413620/3d9adf7d-e0cf-460f-b8e1-c12ccb14ad07)

**Comment by EcCo18** [comment]
Hi tesmez, what deployment method did you use?

**Comment by tesmez** [comment]
Hello EcCol18,

I have used tomcat, postgreSQL, JDK 8 and git cloning through Intelliji code editor

**Comment by EcCo18** [comment]
I would recommend to use docker for setup. Are you familiar with Docker? It is pretty easy to setup.

**Comment by tesmez** [comment]
hi dear,

even i try using docker but it responds the below error

2023-07-11 09:25:09.519] [ERROR] [org.springframework.boot.diagnostics.LoggingFailureAnalysisReporter] - 

***************************
APPLICATION FAILED TO START
***************************

Description:

Failed to configure a DataSource: 'url' attribute is not specified and no embedded datasource could be configured.

Reason: Failed to determine a suitable driver class


Action:

Consider the following:
	If you want an embedded database (H2, HSQL or Derby), please put it on the classpath.
	If you have database settings to be loaded from a particular profile you may need to activate it (the profiles local are currently active).


> Task :burning-okr-app:bootRun FAILED

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':burning-okr-app:bootRun'.
> Process 'command 'C:\Program Files\Java\jdk-1.8\bin\java.exe'' finished with non-zero exit value 1

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.

* Get more help at https://help.gradle.org

Deprecated Gradle features were used in this build, making it incompatible with Gradle 5.0.
See https://docs.gradle.org/4.8.1/userguide/command_line_interface.html#sec:command_line_warnings

BUILD FAILED in 6s
10 actionable tasks: 2 executed, 8 up-to-date

Publishing a build scan to scans.gradle.com requires accepting the Gradle Terms of Service defined at https://gradle.com/terms-of-service. Do you accept these terms? [yes, no]

**Comment by tesmez** [comment]
when i try to build using docker with the command  "docker-compose build"
it returns the below error


 Download https://repo.maven.apache.org/maven2/org/springframework/spring-websocket/5.2.10.RELEASE/spring-websocket-5.2.10.RELEASE.jar
1231.3
1231.3 > Task :burning-okr-model:compileJava FAILED
1231.3 2 actionable tasks: 2 executed
1231.3
1231.3 FAILURE: Build failed with an exception.
1231.3
1231.3 * What went wrong:
1231.3 Could not resolve all files for configuration ':burning-okr-model:compileClasspath'.
1231.3 > Could not download spring-tx.jar (org.springframework:spring-tx:5.0.9.RELEASE)
1231.3    > Could not get resource 'https://repo.maven.apache.org/maven2/org/springframework/spring-tx/5.0.9.RELEASE/spring-tx-5.0.9.RELEASE.jar'.
1231.3       > Could not GET 'https://repo.maven.apache.org/maven2/org/springframework/spring-tx/5.0.9.RELEASE/spring-tx-5.0.9.RELEASE.jar'.
1231.3          > Connect to repo.maven.apache.org:443 [repo.maven.apache.org/199.232.80.215] failed: Read timed out
1231.3 > Could not download spring-expression.jar (org.springframework:spring-expression:5.2.13.RELEASE)
1231.3    > Could not get resource 'https://repo.maven.apache.org/maven2/org/springframework/spring-expression/5.2.13.RELEASE/spring-expression-5.2.13.RELEASE.jar'.
1231.3       > Could not GET 'https://repo.maven.apache.org/maven2/org/springframework/spring-expression/5.2.13.RELEASE/spring-expression-5.2.13.RELEASE.jar'.
1231.3          > Connect to repo.maven.apache.org:443 [repo.maven.apache.org/199.232.80.215] failed: Read timed out
1231.3 > Could not download jwks-rsa.jar (com.auth0:jwks-rsa:0.9.0)
1231.3    > Could not get resource 'https://repo.maven.apache.org/maven2/com/auth0/jwks-rsa/0.9.0/jwks-rsa-0.9.0.jar'.
1231.3       > Could not GET 'https://repo.maven.apache.org/maven2/com/auth0/jwks-rsa/0.9.0/jwks-rsa-0.9.0.jar'.
1231.3          > Connect to repo.maven.apache.org:443 [repo.maven.apache.org/199.232.80.215] failed: Read timed out
1231.3 > Could not download jackson-module-parameter-names.jar (com.fasterxml.jackson.module:jackson-module-parameter-names:2.11.3)
1231.3    > Could not get resource 'https://repo.maven.apache.org/maven2/com/fasterxml/jackson/module/jackson-module-parameter-names/2.11.3/jackson-module-parameter-names-2.11.3.jar'.
1231.3       > Could not GET 'https://repo.maven.apache.org/maven2/com/fasterxml/jackson/module/jackson-module-parameter-names/2.11.3/jackson-module-parameter-names-2.11.3.jar'.
1231.3          > Connect to repo.maven.apache.org:443 [repo.maven.apache.org/199.232.80.215] failed: Read timed out
1231.3 > Could not download micrometer-core.jar (io.micrometer:micrometer-core:1.5.11)
1231.3    > Could not get resource 'https://repo.maven.apache.org/maven2/io/micrometer/micrometer-core/1.5.11/micrometer-core-1.5.11.jar'.
1231.3       > Could not GET 'https://repo.maven.apache.org/maven2/io/micrometer/micrometer-core/1.5.11/micrometer-core-1.5.11.jar'.
1231.3          > Connect to repo.maven.apache.org:443 [repo.maven.apache.org/199.232.80.215] failed: Read timed out
1231.3 > Could not download jboss-transaction-api_1.2_spec.jar (org.jboss.spec.javax.transaction:jboss-transaction-api_1.2_spec:1.0.1.Final)
1231.3    > Could not get resource 'https://repo.maven.apache.org/maven2/org/jboss/spec/javax/transaction/jboss-transaction-api_1.2_spec/1.0.1.Final/jboss-transaction-api_1.2_spec-1.0.1.Final.jar'.
1231.3       > Could not GET 'https://repo.maven.apache.org/maven2/org/jboss/spec/javax/transaction/jboss-transaction-api_1.2_spec/1.0.1.Final/jboss-transaction-api_1.2_spec-1.0.1.Final.jar'.
1231.3          > Connect to repo.maven.apache.org:443 [repo.maven.apache.org/199.232.80.215] failed: Read timed out
1231.3 > Could not download jandex.jar (org.jboss:jandex:2.0.3.Final)
1231.3    > Could not get resource 'https://repo.maven.apache.org/maven2/org/jboss/jandex/2.0.3.Final/jandex-2.0.3.Final.jar'.
1231.3       > Could not GET 'https://repo.maven.apache.org/maven2/org/jboss/jandex/2.0.3.Final/jandex-2.0.3.Final.jar'.
1231.3          > Connect to repo.maven.apache.org:443 [repo.maven.apache.org/199.232.80.215] failed: Read timed out
1231.3 > Could not download dom4j.jar (dom4j:dom4j:1.6.1)
1231.3    > Could not get resource 'https://repo.maven.apache.org/maven2/dom4j/dom4j/1.6.1/dom4j-1.6.1.jar'.
1231.3       > Could not GET 'https://repo.maven.apache.org/maven2/dom4j/dom4j/1.6.1/dom4j-1.6.1.jar'.
1231.3          > Connect to repo.maven.apache.org:443 [repo.maven.apache.org/199.232.80.215] failed: Read timed out
1231.3 > Could not download jackson-annotations.jar (com.fasterxml.jackson.core:jackson-annotations:2.11.4)
1231.3    > Could not get resource 'https://repo.maven.apache.org/maven2/com/fasterxml/jackson/core/jackson-annotations/2.11.4/jackson-annotations-2.11.4.jar'.
1231.3       > Could not GET 'https://repo.maven.apache.org/maven2/com/fasterxml/jackson/core/jackson-annotations/2.11.4/jackson-annotations-2.11.4.jar'.
1231.3          > Connect to repo.maven.apache.org:443 [repo.maven.apache.org/199.232.80.215] failed: Read timed out
1231.3 > Could not download spring-jcl.jar (org.springframework:spring-jcl:5.2.13.RELEASE)
1231.3    > Could not get resource 'https://repo.maven.apache.org/maven2/org/springframework/spring-jcl/5.2.13.RELEASE/spring-jcl-5.2.13.RELEASE.jar'.
1231.3       > Could not GET 'https://repo.maven.apache.org/maven2/org/springframework/spring-jcl/5.2.13.RELEASE/spring-jcl-5.2.13.RELEASE.jar'.
1231.3          > Connect to repo.maven.apache.org:443 [repo.maven.apache.org/199.232.80.215] failed: Read timed out
1231.3 > Could not download spring-boot-starter-tomcat.jar (org.springframework.boot:spring-boot-starter-tomcat:2.3.5.RELEASE)
1231.3    > Could not get resource 'https://repo.maven.apache.org/maven2/org/springframework/boot/spring-boot-starter-tomcat/2.3.5.RELEASE/spring-boot-starter-tomcat-2.3.5.RELEASE.jar'.
1231.3       > Could not GET 'https://repo.maven.apache.org/maven2/org/springframework/boot/spring-boot-starter-tomcat/2.3.5.RELEASE/spring-boot-starter-tomcat-2.3.5.RELEASE.jar'.
1231.3          > Connect to repo.maven.apache.org:443 [repo.maven.apache.org/199.232.80.215] failed: Read timed out
1231.3 > Could not download snakeyaml.jar (org.yaml:snakeyaml:1.26)
1231.3    > Could not get resource 'https://repo.maven.apache.org/maven2/org/yaml/snakeyaml/1.26/snakeyaml-1.26.jar'.
1231.3       > Could not GET 'https://repo.maven.apache.org/maven2/org/yaml/snakeyaml/1.26/snakeyaml-1.26.jar'.
1231.3          > Connect to repo.maven.apache.org:443 [repo.maven.apache.org/199.232.252.215] failed: Read timed out
1231.3 > Could not download LatencyUtils.jar (org.latencyutils:LatencyUtils:2.0.3)
1231.3    > Could not get resource 'https://repo.maven.apache.org/maven2/org/latencyutils/LatencyUtils/2.0.3/LatencyUtils-2.0.3.jar'.
1231.3       > Could not GET 'https://repo.maven.apache.org/maven2/org/latencyutils/LatencyUtils/2.0.3/LatencyUtils-2.0.3.jar'.
1231.3          > Connect to repo.maven.apache.org:443 [repo.maven.apache.org/199.232.252.215] failed: Read timed out
1231.3 > Could not download jsr305.jar (com.google.code.findbugs:jsr305:3.0.2)
1231.3    > Could not get resource 'https://repo.maven.apache.org/maven2/com/google/code/findbugs/jsr305/3.0.2/jsr305-3.0.2.jar'.
1231.3       > Could not GET 'https://repo.maven.apache.org/maven2/com/google/code/findbugs/jsr305/3.0.2/jsr305-3.0.2.jar'.
1231.3          > Connect to repo.maven.apache.org:443 [repo.maven.apache.org/199.232.80.215] failed: Read timed out
1231.3 > Could not download j2objc-annotations.jar (com.google.j2objc:j2objc-annotations:1.1)
1231.3    > Could not get resource 'https://repo.maven.apache.org/maven2/com/google/j2objc/j2objc-annotations/1.1/j2objc-annotations-1.1.jar'.
1231.3       > Could not GET 'https://repo.maven.apache.org/maven2/com/google/j2objc/j2objc-annotations/1.1/j2objc-annotations-1.1.jar'.
1231.3          > Connect to repo.maven.apache.org:443 [repo.maven.apache.org/199.232.80.215] failed: Read timed out
1231.3 > Could not download logback-core.jar (ch.qos.logback:logback-core:1.2.3)
1231.3    > Could not get resource 'https://repo.maven.apache.org/maven2/ch/qos/logback/logback-core/1.2.3/logback-core-1.2.3.jar'.
1231.3       > Could not GET 'https://repo.maven.apache.org/maven2/ch/qos/logback/logback-core/1.2.3/logback-core-1.2.3.jar'.
1231.3          > Connect to repo.maven.apache.org:443 [repo.maven.apache.org/199.232.80.215] failed: Read timed out
1231.3
1231.3 * Try:
1231.3 Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.
1231.3
1231.3 * Get more help at https://help.gradle.org
1231.3
1231.3 BUILD FAILED in 20m 30s
------
failed to solve: process "/bin/sh -c gradle bootJar --no-daemon" did not complete successfully: exit code: 1

**Comment by EcCo18** [comment]
Hi tesmez,

I created you a small guide to get it running:

1. create a folder named burning-okr on your local machine in a directory of your choice.
2. create a file named docker-compose.yaml in this directory and open it in text editor or VS-Code or an editor of your choice
3. click on this [url](https://github.com/BurningOKR/BurningOKR/blob/master/docker/docker-compose-prod.yml)
4. copy the content of the file that you opened in step 3 into your local file that you opened in your editor
5. Safe the file and close the editor
6. create another file in your local burning-okr folder (there should now be exactly one file into it named docker-compose.yaml). The new file have to be named backend.env
7. click on this [url](https://github.com/BurningOKR/BurningOKR/blob/master/docker/backend.env.sample)
8. copy the content of the file opened in step 7 into your local backend.env file
9. edit the local backend.env file in an editor (should now have content into it after step 8)
10. place a # at the **beginning** of the following lines (32-35, 46, 49, 52, 55, 58, 63-64)
11. replace <db-connection-string> in line 20 with jdbc:postgresql://burning-okr-postgres:5432/okr?useSSL=false
12. replace <db-username> in line 23 with okr-user
13. replace <db-password> in line 26 with a password of your choice and remember it will will need it later!
14. replace <local|azure> in line 40 with local
15. create another file in your local burning-okr folder (there should now be two files into it named docker-compose.yaml and backend.env). The new file have to be named postgres.env
16. open the new  postgres.env in an editor
17. click on this [url](https://github.com/BurningOKR/BurningOKR/blob/master/docker/postgres.env.sample)
18. copy the content of the file that you opened in step 17 in your opened file from step 16
19. replace <username> in line 8 with okr-user
20. replace <password> in line 11 with your chosen password from step 13
21. replace <database> in line 14 with okr
22. know check that all local files have been saved
23. you should now have three files in your local folder
24. open a terminal and navigate to your local  folder that you named burning-okr in step 1
25. check that you are in the correct folder by using the command ls -> you should see your three files listed in the terminal
26. know run `docker-compose up -d` in your terminal (if this is not working try running: `docker compose up -d`)
27. your burning-okr instance should now be reachable (wait a few minutes before trying this) when you type http://localhost:80 into your browser

Let me know if this helped.

**Comment by tesmez** [comment]
Hello EcCol18,

first thank you for your follow-up 

i do as per your guidance and localhost is not opening on my browser as yuo can see on the screenshot below

![image](https://github.com/BurningOKR/BurningOKR/assets/66413620/a857a21a-6487-4dc7-a42f-86b41b82c83d)
![image](https://github.com/BurningOKR/BurningOKR/assets/66413620/223042b5-4de8-4d7c-8c15-8467493e63d8)
![image](https://github.com/BurningOKR/BurningOKR/assets/66413620/aaa65036-9c0f-4159-bc3e-be41bdff6250)

**Comment by EcCo18** [comment]
Sorry you need to use http://localhost:4200.

**Comment by tesmez** [comment]
Hello,

it displays nothing still when i try to browse using http://localhost:4200/.
should i create a database name by opening PostgreSQL on my local device?
or what shall i do dear?
![image](https://github.com/BurningOKR/BurningOKR/assets/66413620/06309844-a2ad-4e87-8776-18b56ef59f4c)

**Comment by EcCo18** [comment]
Hi, can you send me the logs from your running Containers? It seems like the Nginx is running but there is some kind of problem.

**Comment by tesmez** [comment]
Hi

please check the below screenshots

Burning-okr Backend logs
------------------------------
2023-07-13 14:09:40 
2023-07-13 14:09:40   .   ____          _            __ _ _
2023-07-13 14:09:40  /\\ / ___'_ __ _ _(_)_ __  __ _ \ \ \ \
2023-07-13 14:09:40 ( ( )\___ | '_ | '_| | '_ \/ _` | \ \ \ \
2023-07-13 14:09:40  \\/  ___)| |_)| | | | | || (_| |  ) ) ) )
2023-07-13 14:09:40   '  |____| .__|_| |_|_| |_\__, | / / / /
2023-07-13 14:09:40  =========|_|==============|___/=/_/_/_/
2023-07-13 14:09:40  :: Spring Boot ::        (v2.0.4.RELEASE)
2023-07-13 14:09:40 
2023-07-13 14:09:40 [2023-07-13 11:09:40.658] [INFO] [org.burningokr.BurningOkrApp] - Starting BurningOkrApp on fb177d9d82cd with PID 1 (/app/burning-okr.jar started by root in /)
2023-07-13 14:09:40 [2023-07-13 11:09:40.669] [INFO] [org.burningokr.BurningOkrApp] - No active profile set, falling back to default profiles: default
2023-07-13 14:09:40 [2023-07-13 11:09:40.743] [INFO] [org.springframework.boot.web.servlet.context.AnnotationConfigServletWebServerApplicationContext] - Refreshing org.springframework.boot.web.servlet.context.AnnotationConfigServletWebServerApplicationContext@97e1986: startup date [Thu Jul 13 11:09:40 UTC 2023]; root of context hierarchy
2023-07-13 14:09:42 [2023-07-13 11:09:42.990] [INFO] [org.springframework.context.support.PostProcessorRegistrationDelegate$BeanPostProcessorChecker] - Bean 'org.springframework.transaction.annotation.ProxyTransactionManagementConfiguration' of type [org.springframework.transaction.annotation.ProxyTransactionManagementConfiguration$$EnhancerBySpringCGLIB$$4fb70b90] is not eligible for getting processed by all BeanPostProcessors (for example: not eligible for auto-proxying)
2023-07-13 14:09:43 [2023-07-13 11:09:43.044] [INFO] [org.springframework.context.support.PostProcessorRegistrationDelegate$BeanPostProcessorChecker] - Bean 'org.springframework.security.config.annotation.configuration.ObjectPostProcessorConfiguration' of type [org.springframework.security.config.annotation.configuration.ObjectPostProcessorConfiguration$$EnhancerBySpringCGLIB$$e864d3ca] is not eligible for getting processed by all BeanPostProcessors (for example: not eligible for auto-proxying)
2023-07-13 14:09:43 [2023-07-13 11:09:43.052] [INFO] [org.springframework.context.support.PostProcessorRegistrationDelegate$BeanPostProcessorChecker] - Bean 'objectPostProcessor' of type [org.springframework.security.config.annotation.configuration.AutowireBeanFactoryObjectPostProcessor] is not eligible for getting processed by all BeanPostProcessors (for example: not eligible for auto-proxying)
2023-07-13 14:09:43 [2023-07-13 11:09:43.056] [INFO] [org.springframework.context.support.PostProcessorRegistrationDelegate$BeanPostProcessorChecker] - Bean 'org.springframework.security.access.expression.method.DefaultMethodSecurityExpressionHandler@23aa363a' of type [org.springframework.security.oauth2.provider.expression.OAuth2MethodSecurityExpressionHandler] is not eligible for getting processed by all BeanPostProcessors (for example: not eligible for auto-proxying)
2023-07-13 14:09:43 [2023-07-13 11:09:43.069] [INFO] [org.springframework.context.support.PostProcessorRegistrationDelegate$BeanPostProcessorChecker] - Bean 'org.springframework.security.config.annotation.method.configuration.GlobalMethodSecurityConfiguration' of type [org.springframework.security.config.annotation.method.configuration.GlobalMethodSecurityConfiguration$$EnhancerBySpringCGLIB$$d39767c] is not eligible for getting processed by all BeanPostProcessors (for example: not eligible for auto-proxying)
2023-07-13 14:09:43 [2023-07-13 11:09:43.086] [INFO] [org.springframework.context.support.PostProcessorRegistrationDelegate$BeanPostProcessorChecker] - Bean 'methodSecurityMetadataSource' of type [org.springframework.security.access.method.DelegatingMethodSecurityMetadataSource] is not eligible for getting processed by all BeanPostProcessors (for example: not eligible for auto-proxying)
2023-07-13 14:09:43 [2023-07-13 11:09:43.611] [INFO] [org.springframework.boot.web.embedded.tomcat.TomcatWebServer] - Tomcat initialized with port(s): 8080 (http)
2023-07-13 14:09:43 [2023-07-13 11:09:43.638] [INFO] [org.apache.coyote.http11.Http11NioProtocol] - Initializing ProtocolHandler ["http-nio-8080"]
2023-07-13 14:09:43 [2023-07-13 11:09:43.656] [INFO] [org.apache.catalina.core.StandardService] - Starting service [Tomcat]
2023-07-13 14:09:43 [2023-07-13 11:09:43.656] [INFO] [org.apache.catalina.core.StandardEngine] - Starting Servlet Engine: Apache Tomcat/8.5.32
2023-07-13 14:09:43 [2023-07-13 11:09:43.674] [INFO] [org.apache.catalina.core.AprLifecycleListener] - The APR based Apache Tomcat Native library which allows optimal performance in production environments was not found on the java.library.path: [/usr/java/packages/lib/amd64:/usr/lib64:/lib64:/lib:/usr/lib]
2023-07-13 14:09:43 [2023-07-13 11:09:43.768] [INFO] [org.apache.catalina.core.ContainerBase.[Tomcat].[localhost].[/]] - Initializing Spring embedded WebApplicationContext
2023-07-13 14:09:43 [2023-07-13 11:09:43.769] [INFO] [org.springframework.web.context.ContextLoader] - Root WebApplicationContext: initialization completed in 3029 ms
2023-07-13 14:09:44 [2023-07-13 11:09:44.186] [ERROR] [org.springframework.boot.web.embedded.tomcat.TomcatStarter] - Error starting Tomcat context. Exception: org.springframework.beans.factory.BeanCreationException. Message: Error creating bean with name 'servletEndpointRegistrar' defined in class path resource [org/springframework/boot/actuate/autoconfigure/endpoint/web/ServletEndpointManagementContextConfiguration$WebMvcServletEndpointManagementContextConfiguration.class]: Bean instantiation via factory method failed; nested exception is org.springframework.beans.BeanInstantiationException: Failed to instantiate [org.springframework.boot.actuate.endpoint.web.ServletEndpointRegistrar]: Factory method 'servletEndpointRegistrar' threw exception; nested exception is org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'healthEndpoint' defined in class path resource [org/springframework/boot/actuate/autoconfigure/health/HealthEndpointConfiguration.class]: Bean instantiation via factory method failed; nested exception is org.springframework.beans.BeanInstantiationException: Failed to instantiate [org.springframework.boot.actuate.health.HealthEndpoint]: Factory method 'healthEndpoint' threw exception; nested exception is org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'org.springframework.boot.actuate.autoconfigure.jdbc.DataSourceHealthIndicatorAutoConfiguration': Bean instantiation via constructor failed; nested exception is org.springframework.beans.BeanInstantiationException: Failed to instantiate [org.springframework.boot.actuate.autoconfigure.jdbc.DataSourceHealthIndicatorAutoConfiguration$$EnhancerBySpringCGLIB$$be9b5d16]: Constructor threw exception; nested exception is org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'dataSource' defined in class path resource [org/springframework/boot/autoconfigure/jdbc/DataSourceConfiguration$Hikari.class]: Bean instantiation via factory method failed; nested exception is org.springframework.beans.BeanInstantiationException: Failed to instantiate [com.zaxxer.hikari.HikariDataSource]: Factory method 'dataSource' threw exception; nested exception is org.springframework.boot.autoconfigure.jdbc.DataSourceProperties$DataSourceBeanCreationException: Failed to determine a suitable driver class
2023-07-13 14:09:44 [2023-07-13 11:09:44.224] [INFO] [org.apache.catalina.core.StandardService] - Stopping service [Tomcat]
2023-07-13 14:09:44 [2023-07-13 11:09:44.267] [WARN] [org.springframework.boot.web.servlet.context.AnnotationConfigServletWebServerApplicationContext] - Exception encountered during context initialization - cancelling refresh attempt: org.springframework.context.ApplicationContextException: Unable to start web server; nested exception is org.springframework.boot.web.server.WebServerException: Unable to start embedded Tomcat
2023-07-13 14:09:44 [2023-07-13 11:09:44.292] [INFO] [org.springframework.boot.autoconfigure.logging.ConditionEvaluationReportLoggingListener] - 
2023-07-13 14:09:44 
2023-07-13 14:09:44 Error starting ApplicationContext. To display the conditions report re-run your application with 'debug' enabled.
2023-07-13 14:09:44 [2023-07-13 11:09:44.297] [ERROR] [org.springframework.boot.diagnostics.LoggingFailureAnalysisReporter] - 
2023-07-13 14:09:44 
2023-07-13 14:09:44 ***************************
2023-07-13 14:09:44 APPLICATION FAILED TO START
2023-07-13 14:09:44 ***************************
2023-07-13 14:09:44 
2023-07-13 14:09:44 Description:
2023-07-13 14:09:44 
2023-07-13 14:09:44 Failed to configure a DataSource: 'url' attribute is not specified and no embedded datasource could be configured.
2023-07-13 14:09:44 
2023-07-13 14:09:44 Reason: Failed to determine a suitable driver class
2023-07-13 14:09:44 
2023-07-13 14:09:44 
2023-07-13 14:09:44 Action:
2023-07-13 14:09:44 
2023-07-13 14:09:44 Consider the following:
2023-07-13 14:09:44     If you want an embedded database (H2, HSQL or Derby), please put it on the classpath.
2023-07-13 14:09:44     If you have database settings to be loaded from a particular profile you may need to activate it (no profiles are currently active).
2023-07-13 14:09:44



Burning-okr frontend logs
----------------------------
2023-07-13 14:11:23 2023/07/13 11:11:19 [emerg] 1#1: host not found in upstream "burning-okr-backend:8080" in /etc/nginx/nginx.conf:15
2023-07-13 14:11:23 nginx: [emerg] host not found in upstream "burning-okr-backend:8080" in /etc/nginx/nginx.conf:15

browser logs
----------------

![image](https://github.com/BurningOKR/BurningOKR/assets/66413620/5a7c6207-a190-4441-9e99-37d35d9f36d3)

**Comment by EcCo18** [comment]
Hi @tesmez, I tried the steps I send you earlier and they worked for me. I saw that GitHubs markdown escaped some words in my text so maybe this is the problem. I will send you some example config files which are working for me. Try them and when you want to use them for production remember to replace the password at line 26 in the backend.env and at line 11 in the postgres.env with the same new password.

postgres.env:
```env
#############################################################
#														    #
# Configuration for the Postgres Database Docker Container. #
#														    #
#############################################################

# The preferred username for the Postgres User of the OKR Tool.
POSTGRES_USER=okr-user

# A secure password for the Postgres User of the OKR Tool.
POSTGRES_PASSWORD=secret

# The name of the database the OKR Tool should use.
POSTGRES_DB=okr
```

backend.env:
```env
##########################################################
#														 #
# Configuration for the Docker Container of the backend. #
#														 #
##########################################################

# The spring profile to use. "prod" is recommended.
SPRING_PROFILES_ACTIVE=prod

# The URL of the Postgres Database.
#
# When using the default Postgres Docker Container, use the URL
# jdbc:postgresql://burning-okr-postgres:5432/<database>?useSSL=false
#
# When using a Postgres database on the host machine, you can use the URL
# jdbc:postgresql://host.docker.internal:5432/<database>?useSSL=false
#
# Otherwise use the following URL Scheme to point to a database at a custom location:
# jdbc:postgresql://<hostname>:<port>/<database>?useSSL=false
SPRING_DATASOURCE_URL=jdbc:postgresql://burning-okr-postgres:5432/okr?useSSL=false

# The username of the Postgres User of the OKR Tool. Should be the same as in the postgres.env.
SPRING_DATASOURCE_USERNAME=okr-user

# The password of the Postgres User of the OKR Tool. Should be the same as in the postgres.env.
SPRING_DATASOURCE_PASSWORD=secret

# The SQL Dialect used by the backend
SPRING_JPA_PROPERTIES_HIBERNATE_DIALECT = org.hibernate.dialect.PostgreSQLDialect

# SMTP Server Configuration
#SPRING_MAIL_USERNAME=<smtp-username>
#SPRING_MAIL_PASSWORD=<smtp-password>
#SPRING_MAIL_HOST=<smtp-host>
#SPRING_MAIL_PORT=<smtp-port>

# The Authentication Mode.
# Use "local" to use the local database for the user management.
# Use "azure" to use an Azure Active Directory with OAuth for the user management.
SYSTEM_CONFIGURATION_AUTH_MODE=local

# The following settings are only for the authentication with the Azure Active Directory.
# You do not need to change anything in the following settings, when you are using the local Authentication Mode.
#
# Azure Ad Issuer
#AZURE_AD_ISSUER=<azure-issuer>

# The Azure Ad ClientId
#SECURITY_OAUTH2_CLIENT_CLIENTID=<azure-clientid>

# The Azure Ad Client secret
#SECURITY_OAUTH2_CLIENT_CLIENTSECRET=<azure-client-secret>

# The Azure Ad Access Token URI
#SECURITY_OAUTH2_CLIENT_ACCESSTOKENURI=<azure-access-token-uri>

# The Azure Ad User Authorization URI
#SECURITY_OAUTH2_CLIENT_USERAUTHORIZATIONURI=<azure-user-authorization-uri>

# The Azure Active Directory groups, that are authorized to use BurningOKR.
# You can add as many groups as you like.
# You must specify at least one group.
#AZURE_AD_AZUREGROUPS_0_NAME=<azure-group-name>
#AZURE_AD_AZUREGROUPS_0_ID=<azure-group-id>
# AZURE_AD_AZUREGROUPS_1_NAME=<second-azure-group-name>
# AZURE_AD_AZUREGROUPS_1_ID=<second-azure-group-id>
# ...

```

**Comment by EcCo18** [comment]
I also updated my comment. Now everything looks correct to me.
[https://github.com/BurningOKR/BurningOKR/issues/261#issuecomment-1632247145](url)

**Comment by tesmez** [comment]
Hi @EcCo18 
I still appreciate for your unlimited support thank you for your willingness and openness

but still the problem exists from my side even i tried your latest comment. still this app not working for me 
i love to use the UI of burning OKR but i am burning-out with this error 

for your hint:
-i am using windows 11 pro
-i am using this docker desktop app https://www.docker.com/products/docker-desktop
please recommend me if either any other tool that i must install to use this app in docker?

![image](https://github.com/BurningOKR/BurningOKR/assets/66413620/a0d7a467-6607-4896-97f1-1da75e8c196b)

**Comment by tesmez** [comment]
Hi @EcCo18 

wow it works for me now
thank you! i am happy on your support. you are one of the best helper Burning OKR team member. 
**i think the installed docker on my desktop was with cache. when i uninstall my desktop docker and reinstall it again your step perfectly works for me also**
now i need to helps only
1. how can i get access and customize burning okr code using docker
2. how to deploy to production for burning okr which is running on my local desk through docker

![image](https://github.com/BurningOKR/BurningOKR/assets/66413620/46aac865-9aa7-4a7e-8820-835d1b3203b3)

**Comment by EcCo18** [comment]
Hi @tesmez. I'm glad I cloud help you. For further support on 3. please open another issue. To modify the source code you have to clone the Git-Repo but you have to create docker containers and other resources by yourself after that. For more information about the development process please refer to the readme.
We can't support a production deployment here on GitHub because it is out of scope for this project and requires knowledge about the infrastructure that is existent, but when you are interested i can give our sales team your contact data to make you an offer for further support on this topic.
